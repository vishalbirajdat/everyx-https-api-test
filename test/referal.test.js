const request = require("supertest");
require("dotenv").config();
const { ethers } = require("ethers");

const API_BASE_URL = process.env.API_BASE_URL || "http://localhost:8800";
const api = request(API_BASE_URL);

const ENDPOINTS = {
  NONCE: "/auth/v2/wallet/nonce",
  VERIFY: "/auth/v2/wallet/verify",
  ME: "/me",
  WALLETS: "/wallets",
};

const REFERRAL_BALANCE = 10;

const VALID_STATUS = [200, 201];

// Enhanced normalizeResponse for snapshotting dynamic fields
const normalizeResponse = (response) => {
  const stableBody = { ...response.body };

  // Mask dynamic fields that will change in each query/test
  if (stableBody.timestamp) stableBody.timestamp = "<DYNAMIC_TIMESTAMP>";
  if (stableBody.sessionId) stableBody.sessionId = "<DYNAMIC_SESSION_ID>";

  // Mask dynamic fields in user object if present
  if (stableBody.user) {
    if (stableBody.user.id) stableBody.user.id = "<DYNAMIC_USER_ID>";
    if (stableBody.user.email) stableBody.user.email = "<DYNAMIC_EMAIL>";
    if (stableBody.user.trader_id) stableBody.user.trader_id = "<DYNAMIC_TRADER_ID>";
    if (stableBody.user.wallet_address) stableBody.user.wallet_address = "<DYNAMIC_WALLET_ADDRESS>";
  }

  // Mask dynamic fields in tracking_data if present
  if (stableBody.tracking_data) {
    if (stableBody.tracking_data.wallet_address)
      stableBody.tracking_data.wallet_address = "<DYNAMIC_WALLET_ADDRESS>";
    if (stableBody.tracking_data.ip)
      stableBody.tracking_data.ip = "<DYNAMIC_IP>";
  }

  // Mask dynamic token if present
  if (stableBody.token) stableBody.token = "<DYNAMIC_TOKEN>";

  // Mask dynamic fields in error/validation if present
  if (stableBody.validation && Array.isArray(stableBody.validation.keys)) {
    stableBody.validation.keys = stableBody.validation.keys.map((k) =>
      typeof k === "string" ? "<DYNAMIC_KEY>" : k
    );
  }

  // Mask dynamic fields in balances/wallets if present
  if (stableBody.balances) {
    Object.keys(stableBody.balances).forEach((k) => {
      if (typeof stableBody.balances[k] === "number") {
        // leave as is, usually 0
      }
    });
    if (stableBody.trader_id) stableBody.trader_id = "<DYNAMIC_TRADER_ID>";
    if (stableBody.wallet_address) stableBody.wallet_address = "<DYNAMIC_WALLET_ADDRESS>";
  }
  if (stableBody.wallets) {
    Object.keys(stableBody.wallets).forEach((k) => {
      if (stableBody.wallets[k].id) stableBody.wallets[k].id = "<DYNAMIC_WALLET_ID>";
      if (stableBody.wallets[k].last_transaction_at !== undefined)
        stableBody.wallets[k].last_transaction_at = "<DYNAMIC_LAST_TX_AT>";
      if (stableBody.wallets[k].wallet_address)
        stableBody.wallets[k].wallet_address = "<DYNAMIC_WALLET_ADDRESS>";
      if (stableBody.wallets[k].trader_id)
        stableBody.wallets[k].trader_id = "<DYNAMIC_TRADER_ID>";
      // Set bonus.balance to 0 or 10 if present
      if (
        stableBody.wallets[k].bonus &&
        typeof stableBody.wallets[k].bonus === "object" &&
        typeof stableBody.wallets[k].bonus.balance === "number"
      ) {
        stableBody.wallets[k].bonus.balance =
          stableBody.wallets[k].bonus.balance === 10 ? 10 : 0;
      }
    });
  }
  // Also mask top-level trader_id and wallet_address if present
  if (stableBody.trader_id) stableBody.trader_id = "<DYNAMIC_TRADER_ID>";
  if (stableBody.wallet_address) stableBody.wallet_address = "<DYNAMIC_WALLET_ADDRESS>";

  // Mask dynamic fields in /me response for referral_code and related fields
  if (stableBody.referral_code) stableBody.referral_code = "<DYNAMIC_REFERRAL_CODE>";
  if (stableBody.referral_code_expires_at) stableBody.referral_code_expires_at = "<DYNAMIC_REFERRAL_CODE_EXPIRES_AT>";
  if (stableBody._id) stableBody._id = "<DYNAMIC_USER_ID>";
  if (stableBody.display_name) stableBody.display_name = "<DYNAMIC_DISPLAY_NAME>";
  if (stableBody.email) stableBody.email = "<DYNAMIC_EMAIL>";

  // Mask dynamic fields in user_level if present
  if (stableBody.user_level && stableBody.user_level.all_levels_progress) {
    Object.values(stableBody.user_level.all_levels_progress).forEach((level) => {
      if (level.progress && typeof level.progress === "object") {
        Object.keys(level.progress).forEach((key) => {
          if (typeof level.progress[key] === "string" && /[0-9a-f]{24}/.test(level.progress[key])) {
            level.progress[key] = "<DYNAMIC>";
          }
        });
      }
    });
  }

  // Set top-level bonus.balance if present
  if (
    stableBody.bonus &&
    typeof stableBody.bonus === "object" &&
    typeof stableBody.bonus.balance === "number"
  ) {
    stableBody.bonus.balance = stableBody.bonus.balance === 10 ? 10 : 0;
  }

  return { status: response.status, body: stableBody };
};

const createWalletAndSign = async () => {
  const wallet = ethers.Wallet.createRandom();
  const nonceResponse = await api.post(ENDPOINTS.NONCE).send({ walletAddress: wallet.address });
  if (nonceResponse.status !== 200) throw new Error("Failed to get nonce");
  const { nonce } = nonceResponse.body;
  const signature = await wallet.signMessage(nonce);
  return { wallet, signature, nonceResponse };
};

const getReferralCodeFromMe = async (token) => {
  const meResponse = await api
    .get(ENDPOINTS.ME)
    .set("authorization", `Bearer ${token}`);
  expect(VALID_STATUS).toContain(meResponse.status);
  // Try common locations for referral code
  const referralCode = meResponse.body.referral_code;
  return referralCode;
};

const getBonusBalance = async (token) => {
  const walletsResponse = await api
    .get(ENDPOINTS.WALLETS)
    .set("authorization", `Bearer ${token}`)
    .query({ include_balances: true });
  expect(VALID_STATUS).toContain(walletsResponse.status);
  const wallets = walletsResponse.body.wallets || walletsResponse.body.data || walletsResponse.body;
  // Try common locations for bonus balance
  let bonus =
    wallets?.bonus?.balance ??
    wallets?.bonus_balance ??
    wallets?.bonusBalance ??
    0;
  bonus = bonus === 10 ? 10 : 0;
  return bonus;
};

describe("Auth V2 Wallet Referral Scenarios", () => {
  let user1, user1Token, user1ReferralCode;
  let user2, user2Token;

  afterAll(() => {
    console.log("✅ All tests completed for Auth V2 Wallet Referral");
  });

  describe("Referral Flow", () => {
    test("Create user1, get referral code, create user2 with referral, user2 gets $10 bonus", async () => {
      // 1. Create user1 (referrer)
      user1 = await createWalletAndSign();
      const verify1 = await api.post(ENDPOINTS.VERIFY).send({
        walletAddress: user1.wallet.address,
        signature: user1.signature,
      });
      expect(VALID_STATUS).toContain(verify1.status);
      expect(verify1.body).toHaveProperty("success", true);
      user1Token = verify1.body.token;
      expect(user1Token).toBeDefined();
      expect(normalizeResponse(verify1)).toMatchSnapshot("user1-verify");

      // 2. Get referral code from /me
      const meResponse = await api
        .get(ENDPOINTS.ME)
        .set("authorization", `Bearer ${user1Token}`);
      expect(VALID_STATUS).toContain(meResponse.status);
      expect(normalizeResponse(meResponse)).toMatchSnapshot("user1-me");
      user1ReferralCode = meResponse.body.referral_code;
      expect(user1ReferralCode).toBeDefined();
      expect(typeof user1ReferralCode).toBe("string");
      expect(user1ReferralCode.length).toBeGreaterThan(0);

      // 3. Create user2 (referee) with referral code
      user2 = await createWalletAndSign();
      const verify2 = await api.post(ENDPOINTS.VERIFY).send({
        walletAddress: user2.wallet.address,
        signature: user2.signature,
        referral_code: user1ReferralCode,
      });
      expect(VALID_STATUS).toContain(verify2.status);
      expect(verify2.body).toHaveProperty("success", true);
      user2Token = verify2.body.token;
      expect(user2Token).toBeDefined();
      expect(normalizeResponse(verify2)).toMatchSnapshot("user2-verify-with-referral");

      // 4. Check user2's bonus balance is $10
      const walletsResponse = await api
        .get(ENDPOINTS.WALLETS)
        .set("authorization", `Bearer ${user2Token}`)
        .query({ include_balances: true });
      expect(VALID_STATUS).toContain(walletsResponse.status);
      expect(normalizeResponse(walletsResponse)).toMatchSnapshot("user2-wallets-bonus");
      const wallets = walletsResponse.body.wallets || walletsResponse.body.data || walletsResponse.body;
      let user2Bonus =
        wallets?.bonus?.balance ??
        wallets?.bonus_balance ??
        wallets?.bonusBalance ??
        0;
      user2Bonus = user2Bonus === 10 ? 10 : 0;
      expect(user2Bonus).toBe(REFERRAL_BALANCE);
    });

    test("Create user3 with non-existent referral code → success, no bonus", async () => {
      const user3 = await createWalletAndSign();
      const verify3 = await api.post(ENDPOINTS.VERIFY).send({
        walletAddress: user3.wallet.address,
        signature: user3.signature,
        referral_code: "NONEXISTENTREFERRALCODE123",
      });
      expect(VALID_STATUS).toContain(verify3.status);
      expect(verify3.body).toHaveProperty("success", true);
      expect(normalizeResponse(verify3)).toMatchSnapshot("user3-verify-nonexistent-referral");
      const user3Token = verify3.body.token;
      expect(user3Token).toBeDefined();

      const walletsResponse = await api
        .get(ENDPOINTS.WALLETS)
        .set("authorization", `Bearer ${user3Token}`)
        .query({ include_balances: true });
      expect(VALID_STATUS).toContain(walletsResponse.status);
      expect(normalizeResponse(walletsResponse)).toMatchSnapshot("user3-wallets-nonexistent-referral");
      const wallets = walletsResponse.body.wallets || walletsResponse.body.data || walletsResponse.body;
      let user3Bonus =
        wallets?.bonus?.balance ??
        wallets?.bonus_balance ??
        wallets?.bonusBalance ??
        0;
      user3Bonus = user3Bonus === 10 ? 10 : 0;
      expect(user3Bonus).toBe(0);
    });

    test("Create user4 with invalid referral code → success, no bonus", async () => {
      const user4 = await createWalletAndSign();
      const verify4 = await api.post(ENDPOINTS.VERIFY).send({
        walletAddress: user4.wallet.address,
        signature: user4.signature,
        referral_code: "INVALID_CODE_!@#",
      });
      expect(VALID_STATUS).toContain(verify4.status);
      expect(verify4.body).toHaveProperty("success", true);
      expect(normalizeResponse(verify4)).toMatchSnapshot("user4-verify-invalid-referral");
      const user4Token = verify4.body.token;
      expect(user4Token).toBeDefined();

      const walletsResponse = await api
        .get(ENDPOINTS.WALLETS)
        .set("authorization", `Bearer ${user4Token}`)
        .query({ include_balances: true });
      expect(VALID_STATUS).toContain(walletsResponse.status);
      expect(normalizeResponse(walletsResponse)).toMatchSnapshot("user4-wallets-invalid-referral");
      const wallets = walletsResponse.body.wallets || walletsResponse.body.data || walletsResponse.body;
      let user4Bonus =
        wallets?.bonus?.balance ??
        wallets?.bonus_balance ??
        wallets?.bonusBalance ??
        0;
      user4Bonus = user4Bonus === 10 ? 10 : 0;
      expect(user4Bonus).toBe(0);
    });

    test("Create user5 with no referral code → success, no bonus", async () => {
      const user5 = await createWalletAndSign();
      const verify5 = await api.post(ENDPOINTS.VERIFY).send({
        walletAddress: user5.wallet.address,
        signature: user5.signature,
      });
      expect(VALID_STATUS).toContain(verify5.status);
      expect(verify5.body).toHaveProperty("success", true);
      expect(normalizeResponse(verify5)).toMatchSnapshot("user5-verify-no-referral");
      const user5Token = verify5.body.token;
      expect(user5Token).toBeDefined();

      const walletsResponse = await api
        .get(ENDPOINTS.WALLETS)
        .set("authorization", `Bearer ${user5Token}`)
        .query({ include_balances: true });
      expect(VALID_STATUS).toContain(walletsResponse.status);
      expect(normalizeResponse(walletsResponse)).toMatchSnapshot("user5-wallets-no-referral");
      const wallets = walletsResponse.body.wallets || walletsResponse.body.data || walletsResponse.body;
      let user5Bonus =
        wallets?.bonus?.balance ??
        wallets?.bonus_balance ??
        wallets?.bonusBalance ??
        0;
      user5Bonus = user5Bonus === 10 ? 10 : 0;
      expect(user5Bonus).toBe(0);
    });

    test("Create user6 with empty referral code → fail", async () => {
      const user6 = await createWalletAndSign();
      const verify6 = await api.post(ENDPOINTS.VERIFY).send({
        walletAddress: user6.wallet.address,
        signature: user6.signature,
        referral_code: "",
      });
      expect([400, 404]).toContain(verify6.status);
      expect(normalizeResponse(verify6)).toMatchSnapshot("user6-verify-empty-referral");
    });
  });
});


//// here if referal falls then no chnages has added and in sometime bonus not recived