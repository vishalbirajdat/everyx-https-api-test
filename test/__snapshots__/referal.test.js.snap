// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user1, get referral code, create user2 with referral, user2 gets $10 bonus: user1-me 1`] = `
{
  "body": {
    "_id": "<DYNAMIC_USER_ID>",
    "display_name": "<DYNAMIC_DISPLAY_NAME>",
    "email": "<DYNAMIC_EMAIL>",
    "email_verified": false,
    "full_name": "",
    "identity_verified": false,
    "isUpgraded": false,
    "is_new": true,
    "phone_verified": false,
    "referral_code": "<DYNAMIC_REFERRAL_CODE>",
    "referral_code_expires_at": "<DYNAMIC_REFERRAL_CODE_EXPIRES_AT>",
    "referral_code_quota": 9999999,
    "user_level": {
      "all_levels_progress": {
        "0": {
          "coratio": 0.3,
          "is_completed": true,
          "is_current": true,
          "level": 0,
          "name": "Untrusted (Rookie Forecaster)",
          "progress": {
            "signup": true,
          },
          "requirements": [],
        },
        "1": {
          "coratio": 0.25,
          "is_completed": false,
          "is_current": false,
          "level": 1,
          "name": "Noob (Predictor in Training)",
          "progress": {
            "deposit": 0,
            "email_verified": false,
            "phone_verified": false,
            "traded_amount": 0,
          },
          "requirements": [
            "Verify email address",
            "Verify phone number",
            "Make a deposit of at least $1",
            "Trade $1.00 more in non-bonus money",
          ],
        },
        "2": {
          "coratio": 0.2,
          "is_completed": false,
          "is_current": false,
          "level": 2,
          "name": "Regular (Seasoned Speculator)",
          "progress": {
            "kyc_verified": false,
            "level_1_requirement": false,
            "traded_amount": 0,
          },
          "requirements": [
            "Complete KYC verification",
            "Trade $100.00 more in non-bonus money (last 90 days)",
          ],
        },
        "3": {
          "coratio": 0.1,
          "is_completed": false,
          "is_current": false,
          "level": 3,
          "name": "Trusted Regular (Market Oracle)",
          "progress": {
            "level_2_requirement": false,
            "traded_amount": 0,
          },
          "requirements": [
            "Trade $10000.00 more in non-bonus money (last 90 days)",
          ],
        },
      },
      "bypass_enabled": false,
      "coratio": 0.3,
      "current_level": 0,
      "level_name": "Untrusted (Rookie Forecaster)",
      "perks": {
        "can_browse_events": true,
        "can_deposit_credit_card": false,
        "can_deposit_crypto": true,
        "can_receive_bonuses": true,
        "can_refer_friends": true,
        "can_trade_regular": true,
        "can_trade_vip_events": false,
        "can_use_bonuses": false,
        "can_withdraw": true,
      },
    },
    "wallet_address": "<DYNAMIC_WALLET_ADDRESS>",
    "withdrawal_limits": {
      "limit": 200,
      "period": "monthly",
      "remaining": 200,
      "reset_date": "2025-10-31T18:30:00.000Z",
      "used": 0,
    },
  },
  "status": 200,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user1, get referral code, create user2 with referral, user2 gets $10 bonus: user1-verify 1`] = `
{
  "body": {
    "success": true,
    "token": "<DYNAMIC_TOKEN>",
    "tracking_data": {
      "ip": "<DYNAMIC_IP>",
      "wallet_address": "<DYNAMIC_WALLET_ADDRESS>",
    },
    "user": {
      "email": "<DYNAMIC_EMAIL>",
      "id": "<DYNAMIC_USER_ID>",
    },
    "user_action": "signup",
  },
  "status": 201,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user1, get referral code, create user2 with referral, user2 gets $10 bonus: user2-verify-with-referral 1`] = `
{
  "body": {
    "success": true,
    "token": "<DYNAMIC_TOKEN>",
    "tracking_data": {
      "ip": "<DYNAMIC_IP>",
      "wallet_address": "<DYNAMIC_WALLET_ADDRESS>",
    },
    "user": {
      "email": "<DYNAMIC_EMAIL>",
      "id": "<DYNAMIC_USER_ID>",
    },
    "user_action": "signup",
  },
  "status": 201,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user1, get referral code, create user2 with referral, user2 gets $10 bonus: user2-wallets-bonus 1`] = `
{
  "body": {
    "balances": {
      "bonus": 10,
      "profit": 0,
      "topup": 0,
      "total": 10,
      "withdrawable": 0,
    },
    "success": true,
    "trader_id": "<DYNAMIC_TRADER_ID>",
    "wallets": {
      "bonus": {
        "balance": 10,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "profit": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "topup": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
    },
  },
  "status": 200,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user3 with non-existent referral code → success, no bonus: user3-verify-nonexistent-referral 1`] = `
{
  "body": {
    "success": true,
    "token": "<DYNAMIC_TOKEN>",
    "tracking_data": {
      "ip": "<DYNAMIC_IP>",
      "wallet_address": "<DYNAMIC_WALLET_ADDRESS>",
    },
    "user": {
      "email": "<DYNAMIC_EMAIL>",
      "id": "<DYNAMIC_USER_ID>",
    },
    "user_action": "signup",
  },
  "status": 201,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user3 with non-existent referral code → success, no bonus: user3-wallets-nonexistent-referral 1`] = `
{
  "body": {
    "balances": {
      "bonus": 0,
      "profit": 0,
      "topup": 0,
      "total": 0,
      "withdrawable": 0,
    },
    "success": true,
    "trader_id": "<DYNAMIC_TRADER_ID>",
    "wallets": {
      "bonus": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "profit": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "topup": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
    },
  },
  "status": 200,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user4 with invalid referral code → success, no bonus: user4-verify-invalid-referral 1`] = `
{
  "body": {
    "success": true,
    "token": "<DYNAMIC_TOKEN>",
    "tracking_data": {
      "ip": "<DYNAMIC_IP>",
      "wallet_address": "<DYNAMIC_WALLET_ADDRESS>",
    },
    "user": {
      "email": "<DYNAMIC_EMAIL>",
      "id": "<DYNAMIC_USER_ID>",
    },
    "user_action": "signup",
  },
  "status": 201,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user4 with invalid referral code → success, no bonus: user4-wallets-invalid-referral 1`] = `
{
  "body": {
    "balances": {
      "bonus": 0,
      "profit": 0,
      "topup": 0,
      "total": 0,
      "withdrawable": 0,
    },
    "success": true,
    "trader_id": "<DYNAMIC_TRADER_ID>",
    "wallets": {
      "bonus": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "profit": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "topup": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
    },
  },
  "status": 200,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user5 with no referral code → success, no bonus: user5-verify-no-referral 1`] = `
{
  "body": {
    "success": true,
    "token": "<DYNAMIC_TOKEN>",
    "tracking_data": {
      "ip": "<DYNAMIC_IP>",
      "wallet_address": "<DYNAMIC_WALLET_ADDRESS>",
    },
    "user": {
      "email": "<DYNAMIC_EMAIL>",
      "id": "<DYNAMIC_USER_ID>",
    },
    "user_action": "signup",
  },
  "status": 201,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user5 with no referral code → success, no bonus: user5-wallets-no-referral 1`] = `
{
  "body": {
    "balances": {
      "bonus": 0,
      "profit": 0,
      "topup": 0,
      "total": 0,
      "withdrawable": 0,
    },
    "success": true,
    "trader_id": "<DYNAMIC_TRADER_ID>",
    "wallets": {
      "bonus": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "profit": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
      "topup": {
        "balance": 0,
        "currency": "USDT",
        "id": "<DYNAMIC_WALLET_ID>",
        "last_transaction_at": "<DYNAMIC_LAST_TX_AT>",
      },
    },
  },
  "status": 200,
}
`;

exports[`Auth V2 Wallet Referral Scenarios Referral Flow Create user6 with empty referral code → fail: user6-verify-empty-referral 1`] = `
{
  "body": {
    "error": "Bad Request",
    "message": ""referral_code" is not allowed to be empty",
    "statusCode": 400,
    "validation": {
      "keys": [
        "<DYNAMIC_KEY>",
      ],
      "source": "payload",
    },
  },
  "status": 400,
}
`;
